docs/modules/ROOT/pages/advanced-scripting.adoc

```diff
@@ -0,0 +1,749 @@
+[source,bash]
+----
+clc script output1.star
+----
+
+Outputs:
+----
+There are 3 fruits.
+Unnecessary output lines will dissapear when you use -q
+
+----------------
+ Index | Fruit
+----------------
+     0 | apple
+     1 | grapes
+     2 | pear
+----------------
+    OK Returned 3 row(s).
+----
+
+Let's now use the `-q` flag to suppress unnecessary output:
+[source,bash]
+----
+clc script output1.star -q
+----
+
+Outputs:
+----
+----------------
+ Index | Fruit
+----------------
+     0 | apple
+     1 | grapes
+     2 | pear
+----------------
+----
+
+As expected, unnecessary output was not produced.
+Using a different output format also works:
+[source,bash]
+----
+clc script output1.star -q -f csv
+----
+
+Outputs:
+----
+Index,Fruit
+0,apple
+1,grapes
+2,pear
+----
+
+Separating unnecessary and structured output makes it much easier to extracts the result using a pipe.
+This example uses the link:https://jqlang.github.io/jq/[jq] utility:
+:
+[source,bash]
+----
+clc script output1.star -q -f json | jq '.Fruit'
+----
+Outputs:
+----
+"apple"
+"grapes"
+"pear"
+----
+
+
+Passing keyword arguments to `output` for column label and value may sometimes be limiting.
+Since only valid Starlark identifiers can be used as the column label.
+For example, running the following script will return an error:
+[source,python]
+----
+def main():
+    output(1=2)
+----
+Because of that, CLC also supports passing a dictionary to the `output` function.
+In that case, the dictionary must be passed as a positional argument and it must be the only argument to `output`.
+You can rewrite the following as follows and it will run just fine:
+[source,python]
+----
+def main():
+    output({1: 2})
+----
+
+The output function handles rows with missing values just fine.
+Here's a script that demonstrate that.
+Save it as `output2.star`:
+[source,python]
+----
+def main():
+    for i in range(3):
+        output({i: "OK"})
+----
+Running it outputs the following:
+----
+--------------
+ 0  | 1  | 2
+--------------
+ OK | -  | -
+ -  | OK | -
+ -  | -  | OK
+--------------
+----
+Different output formats handle missing values differently.
+JSON printer just skips the missing values:
+[source,bash]
+----
+clc script output2.star -f json
+----
+Outputs:
+----
+{"0":"OK"}
+{"1":"OK"}
+{"2":"OK"}
+----
+
+== Error Handling
+
+Whenever an error occurs, CLC stops the script and prints an error message.
+This is usually the safest way to handle the error.
+But there are times when you would like to try something which can result in an error and handle it without giving back the control.
+
+The `script` command supports `the --ignore-errors` flag that suppresses errors while running the script.
+You can use the `last_error()` function to get the raised error and act on it.
+
+Let's demonstrate that with an example using the `data_value` function.
+Passing a string to `data_value` ends with an error, but we are going to cover "why?" in another section.
```

Comment:
with an error; we cover why that is in [link to section].

Reply:
I changed that sentence to:

> Passing a string to `data_value` ends with an error; since that function expects an argument of type `data`.